parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -648,7 +689,8 @@ public void writeDataPageHeader(
       org.apache.parquet.column.Encoding dlEncoding,
       org.apache.parquet.column.Encoding valuesEncoding,
       OutputStream to) throws IOException {
-    writePageHeader(newDataPageHeader(uncompressedSize,
+    writePageHeader(newDataPageHeader(createdBy,
```

Comment:
I don't know how I ended up making write side changes :). I have added the checks to read path and have reverted write side changes.